keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -883,20 +889,43 @@ is disabled [Disconnected node when taint-manager is disabled](#disconnected-nod
 However, as disabling taint-manager is deprecated it is not a concern for this
 KEP.
 
-For Alpha, we implemented a fix for this issue by setting the pod phase as `Failed`
-in podgc.
-
-For Beta, we are going to simplify the code in job controller responsible for
-detecting pod failures. For this purpose we need to make sure that pods stuck
-in the pending phase which are terminating (with set deletionTimestamp) are
-eventually marked as failed. Such pods may either be not scheduled
-(for example when they are unschedulable due to node affinity configuration for
-which no node exists in the cluster) or scheduled (for example when they fail
-to pull the the docker image). In case of non-scheduled pods they will be moved
-to the `Failed` phase by the PodGC controller as it marks as failed all
-non-scheduled terminating pods (after the Alpha changes for this feature, see
-above). We plan to extend the PodGC controller to also move to the failed phase
-(and delete) scheduled pods in the pending state which are terminating.
+A scheduled pod may also get stuck in a `Pending` phase (for example due to an
+invalid image reference). In that case, as long as the pod is not Terminating it
+could be a transient issue and Kubelet can make progress after retrying. However,
+once such a Pod (in a `Pending` phase) is also terminating, Kubelet is not going
+to attempt starting of the pod.
+
+Such cases are problematic in case of Jobs with pod failure policies, which we want
+to match only against pods in terminal state, to ensure their state is not
+modified while Job controller matches them against the policy.
+
+In order to ensure that Job controller only considers pods in terminal phase we
+need to make sure pods in the following states are transitioned into `Failed`
+phase:
+1. a pod is in Running phase and orphaned
+2. a pod is in Pending phase, terminating and unscheduled
```

Comment:
is oxford comma ommitted here on purpose? This comes to the question above on expanding phases description

Reply:
It wasn’t deliberate, I think it might be simpler to understand if commas are added in between all terms. Updated accordingly.

Reply:
Rephrased to account for the feedback, PTAL.